### PR TITLE
[System.Data] referencesource -> corefx

### PR DESCRIPTION
**Pull request by EgorBo**
I've replaced almost all referencesource links in corefx.common.source to corefx.
API differences (gui-compare):
-  DataSet, DataRow, DataView, etc types now don't have some designer (WinForms?) specific attributes:
```
[System.ComponentModel.EditorAttribute]
[System.ComponentModel.Designer]
[System.ComponentModel.ToolboxItem]
```
XmlDataDocument used to be public in Mono but in corefx it's internal. I create a 'public' copy of it in corefx/XmlDataDocument.cs to keep the same API we had. However, this type has [Obsolete] attribute on it .

### CODE REVIEW

File: mcs/class/referencesource/System.Data/System/Data/Common/DbDataAdapter.cs [hunk 1]
```diff
@@ -21,8 +21,8 @@ namespace System.Data.Common {
     public abstract class DbDataAdapter : DataAdapter, IDbDataAdapter, ICloneable { // V1.0.3300, MDAC 69629
         public const string DefaultSourceTableName = "Table"; // V1.0.3300
 
-        internal static readonly object ParameterValueNonNullValue = 0;
-        internal static readonly object ParameterValueNullValue = 1;
+        internal static readonly object s_parameterValueNonNullValue = 0;
```
Comment (marek-safar) [inline mcs/class/referencesource/System.Data/System/Data/Common/DbDataAdapter.cs]: Why did you make this change?

File: mcs/class/referencesource/System.Data/System/Data/Common/DBDataPermissionAttribute.cs [hunk 2]
```diff
@@ -90,14 +90,4 @@ public abstract class DBDataPermissionAttribute : CodeAccessSecurityAttribute {
             return (null != _restrictions);
         }
     }
-}
-
```
Comment (marek-safar) [inline mcs/class/referencesource/System.Data/System/Data/Common/DBDataPermissionAttribute.cs]: Should be #if !MONO instead

File: mcs/class/System.Data/corefx/XmlDataDocument.cs [hunk 3]
```diff
@@ -0,0 +1,3138 @@
+// Licensed to the .NET Foundation under one or more agreements.
```
Comment (marek-safar) [inline mcs/class/System.Data/corefx/XmlDataDocument.cs]: Don't copy files, do #if MONO with your changes in our corefx fork instead

Comment (EgorBo) [inline mcs/class/System.Data/corefx/XmlDataDocument.cs]: @marek-safar ok, I thought it should be clean from changes, but makes sense.

File: mcs/class/referencesource/System.Data/System/Data/Common/DbDataAdapter.cs (hunk 1 above)
Comment (EgorBo) [inline mcs/class/referencesource/System.Data/System/Data/Common/DbDataAdapter.cs]: It was an intermediate commit to compile build, now this file is not used at all - I am going to remove the whole mcs/class/referencesource/System.Data/ folder once I finish or I can remove files with each change..

Comment (marek-safar) [inline mcs/class/referencesource/System.Data/System/Data/Common/DbDataAdapter.cs]: right, so this is a revert to previous version

**Comment by EgorBo** [comment]
@marek-safar 
fixed.
I've updated external/corefx submodule (Dec 22 2016 -> Jan 28 2017)
things I had to do:
1) Add some missing references, members
2) Add DataSetSchemaImporterExtension.cs
corefx doesn't have it and it used to be inside /referencesource/__/DataSet.cs file
3) Add System.IO.Compression/corefx/Interop.cs
https://github.com/mono/corefx/commit/c656446f01db4b902b0c06b19551cc30d9dd9c6b#diff-6ce4d5e9b4023333d4408a947dd214d5L39 -- old impl Mono used to use was deleted. I saved it to the Interop.cs file

**Comment by EgorBo** [comment]
Ok, since there is already a PR "bump corefx" I've rolled back my commits

**Pull request merged**